ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/data/PaywallViewModel.kt

```diff
@@ -162,38 +160,56 @@ internal class PaywallViewModelImpl(
         }
     }
 
-    private fun updateOffering() {
+    private fun updateState() {
         viewModelScope.launch {
             try {
-                val offerings = Purchases.sharedInstance.awaitOfferings()
-                val currentOffering = options.offeringSelection.offeringIdentifier?.let { offerings[it] }
-                    ?: offerings.current
+                var currentOffering = options.offeringSelection.offering
+                if (currentOffering == null) {
+                    val offerings = Purchases.sharedInstance.awaitOfferings()
+                    options.offeringSelection.offeringIdentifier?.let { offerings[it] }
+                        ?: offerings.current
+                }
+
                 if (currentOffering == null) {
                     _state.value = PaywallViewState.Error("No offering or current offering")
                 } else {
-                    _state.value = currentOffering.calculateState()
+                    _state.value = calculateState(
+                        currentOffering,
+                        Purchases.sharedInstance.awaitCustomerInfo(),
+                        _colorScheme.value,
+                    )
                 }
             } catch (e: PurchasesException) {
                 _state.value = PaywallViewState.Error(e.toString())
             }
         }
     }
 
-    @Suppress("TooGenericExceptionCaught")
-    private fun Offering.calculateState(): PaywallViewState {
-        if (availablePackages.isEmpty()) {
-            return PaywallViewState.Error("No packages available")
-        }
-        val (displayablePaywall, template, error) = validatedPaywall(_colorScheme.value)
+    private fun getCurrentLocaleList(): LocaleListCompat {
+        return LocaleListCompat.getDefault()
+    }
+}
 
-        error?.let { validationError ->
-            Logger.w(validationError.associatedErrorString(this))
-            Logger.w(PaywallValidationErrorStrings.DISPLAYING_DEFAULT)
-        }
-        return toPaywallViewState(variableDataProvider, mode, displayablePaywall, template)
+@Suppress("TooGenericExceptionCaught")
```

Comment:
I think this line is probably not needed anymore? Since we are not catching any errors.